src/Adapter/Redis.php

```diff
@@ -36,31 +36,43 @@ public function setErrorCount(string $service = 'default', int $value = 0): void
     /**
      * @inheritdoc
      */
-    public function getErrorCount(string $service = 'default'): int
+    public function decrErrorCount(string $service = 'default', int $value = 1): void
     {
-        return $this->getKey(
-            sprintf('%s.%s.error_count', $this->prefix, $service)
+        $this->redis->decrBy(
+            sprintf('%s.%s.error_count', $this->prefix, $service),
+            $value
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setErrorCount(string $service = 'default', int $value = 0): void
+    {
+        $this->redis->set(
+            sprintf('%s.%s.error_count', $this->prefix, $service),
+            $value
         );
     }
 
     /**
      * @inheritdoc
      */
-    public function getLastCheck(string $service = 'default'): int
+    public function getErrorCount(string $service = 'default'): int
     {
         return $this->getKey(
-            sprintf('%s.%s.last_check', $this->prefix, $service)
+            sprintf('%s.%s.error_count', $this->prefix, $service)
         );
     }
 
     /**
      * @inheritdoc
      */
-    public function updateLastCheck(string $service = 'default'): int
+    public function updateLastCheck(string $service = 'default', int $timeout): void
     {
-        $this->redis->set(
-            sprintf('%s.%s.last_check', $this->prefix, $service),
-            time()
+        $this->redis->expire(
```

Comment:
This is clever, but I cannot remove the key and put the error count back on 0.

This will allow the service to be flooded when it was not yet recovered.